lib/streaming-pipeline.js

```diff
@@ -6,23 +6,15 @@ const logger = require('util').debuglog('riff');
 const InputUnmarshaller = require('./input-unmarshaller');
 const {guardWithTimeout} = require('./async');
 
-const toObject = (parameters) => {
+const toObject = (parameters, streamNames) => {
     const result = {};
     for (let i = 0; i < parameters.length; i++) {
-        result[`${i}`] = parameters[i];
+        const stream = parameters[i];
+        result[`${i}`] = stream;
+        result[streamNames[i]] = stream;
```

Comment:
Stream names and indices can collide, for example, a stream named `0`.

Reply:
I'm willing to let it collide in that case. Would that be an actual issue?

Reply:
Ambiguity is bad, especially for things that may be reconfigured in production.

What if we did something more like:
```
const result = parameters.reduce((result, stream, i) => {
  result[streamNames[i]] = stream
  return result
}, { $order: [...streamNames] })
```

- the dominant index is the stream name
- the order is available via `$order` for any function that cares about the ordering
  - `$order` is not a valid stream name, so there is no risk of collision

Reply:
I favored backwards compatibility, which seemed more important to me than collision (and saved me some time migrating internal tests).
If supporting backwards compatibility is not important (yet), then I'll happily follow the approach you describe.

Reply:
The streaming interface has been defined as experimental. There's no need to sacrifice a better user experience for the sack of backwards compatibility.